syllabus/syllabus-app/src/java/org/sakaiproject/tool/syllabus/util/DateUtil.java

```diff
@@ -0,0 +1,104 @@
+/**********************************************************************************
+ * $URL$
+ * $Id$
+ **********************************************************************************
+ *
+ * Copyright (c) 2008 The Sakai Foundation
+ *
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **********************************************************************************/
+package org.sakaiproject.tool.syllabus.util;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.Locale;
+
+/**
+ * Performs date validation respecting i18n.<br>
+ * <b>Note:</b> This class does not support "hi_IN", "ja_JP_JP" and "th_TH" locales.
+ */
+public final class DateUtil {
```

Comment:
Something like this probably makes sense in a more global location. Otherwise will end up with many tools having their own version of it which violates DRY.